pkg/timescale/write.go

```diff
@@ -1,15 +1,50 @@
 package timescale
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"strconv"
 
 	"github.com/jackc/pgx"
+	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/max-bytes/metrics-receiver/pkg/config"
 	"github.com/max-bytes/metrics-receiver/pkg/general"
 )
 
+var timescalePools map[string]*pgxpool.Pool
+
+func InitConnPools(cfg []config.OutputTimescale) error {
+
+	timescalePools = make(map[string]*pgxpool.Pool)
+
+	for _, output := range cfg {
+
+		c, errC := pgxpool.ParseConfig(output.Connection)
+
+		if errC != nil {
+			return fmt.Errorf("Unable to parse config for timescale database: %v\n", errC)
+		}
+
+		timescaleDbPool, err := pgxpool.ConnectConfig(context.Background(), c)
```

Comment:
Does this already connect to the database? What happens when (at least one of) the databases are not reachable on init?

In general, can the pool recover when a database connection disconnects during runtime, but then later becomes reachable again?